lib/ProductOpener/RequestStats.pm

```diff
@@ -0,0 +1,80 @@
+
+=head1 NAME
+
+ProductOpener::RequestStats - measure and log some stats (e.g. execution time, database request time) for requests
+
+=head1 DESCRIPTION
+
+
+=cut
+
+package ProductOpener::RequestStats;
+
+use ProductOpener::Config qw/:all/;
+use ProductOpener::PerlStandards;
+use Exporter qw< import >;
+
+BEGIN {
+	use vars qw(@ISA @EXPORT_OK %EXPORT_TAGS);
+	@EXPORT_OK = qw(
+		&init_request_stats
+		&set_request_stats_value
+		&set_request_stats_time_value
+		&log_request_stats
+	);    # symbols to export on request
+	%EXPORT_TAGS = (all => [@EXPORT_OK]);
+}
+
+use vars @EXPORT_OK;
+
+use Log::Any qw/$log/;
+use Time::HiRes;
+
+# Specific logger to log request stats
+our $requeststats_log = Log::Any->get_logger(category => 'requeststats');
+
+sub init_request_stats() {
+
+	my $stats_ref = {};
+	set_request_stats_time_value($stats_ref, "request_start");
+	return $stats_ref;
+}
+
+sub set_request_stats_value($stats_ref, $key, $value) {
+	$stats_ref->{$key} = $value;
+	return;
+}
+
+sub set_request_stats_time_value($stats_ref, $key) {
+	$stats_ref->{$key} = Time::HiRes::time();
```

Comment:
nit: I would have added a is_start boolean parameter to add _start / _end to the key. So the logic of _start / _end is contained in this module (as it is used by log_request_stats).

Reply:
I'm changing it to have 2 functions set_request_stats_time_start() (and _end() ), to make it clearer when reading the calling code than having a boolean parameter.